Cooking overhaul

phil235 opened the pull request:
adding tablecraft food recipes, customizable food, new food ingredients (from bay)
- converted the microwave recipes to tablecraft recipes
- Added customizable food (inspired by iamgoofball's PR)
- added food ingredients (from baystation), you can used knife/rollingpin on some of them.
- tray can now be used to load food into the microwave or reagentgrinder.
- changing food recipes a bit (replacing stuff with the new food ingredients)
- repurposed the microwave to actually cook certain food items, and added a power setting to it.
- bowl is no longer trash but a reagent_containers/glass that is used in salad&soup recipes
- Changed the hunger system a bit, sugar becomes a normal reagent again. Adding a junkiness var to foods to drain mob satiety.
- milk carton is now a condiment bottle
- slight buff to vitamin

This PR will make cooking not require using the wiki all the time and will make tablecrafting more visible.

![](http://i.imgur.com/PWpfWaG.png)

phil235 committed:
Cooking overhaul: adding tablecraft food recipes, customizable food, food ingredients (from bay)

- converted the microwave recipes to tablecraft recipes
- Added customizable food (thanks iamgoofball)
- added food ingredients (from baystation)
- changing food recipes a bit (replacing stuff with the new food ingredients)
- repurpose the microwave to actually cook certain food items.
- bowl is no longer trash but a reagent_containers/glass that is used in salad/soup recipes
- Changed the hunger system a bit, sugar becomes a normal reagent again. Adding a junkiness var to foods to drain mob satiety.
- milk carton is now a condiment bottle
- slight buff to vitamin

phil235 committed:
- Some fixes
- Adding power var to microwave to reduce microwaving time (with risk of failure)
- Giving unique color to snacks for filling purposes
- cake batter and pie dough can also be made with soy milk

phil235 committed:
Updating the cook book. Finishing giving filling color to growns. More fixes

phil235 committed:
Merge branch 'master' of https://github.com/tgstation/-tg-station into CookingOverhaul

phil235 committed:
repo update

phil235 committed:
Yet more fixes.

Iamgoofball commented on code/modules/food&drinks/food/customizables.dm:
```diff
@@ -0,0 +1,400 @@
+	bitesize = 3
+	w_class = 3
+
+	var/ingMax = 12
```
>max of 12

make it 50

Iamgoofball commented:
port the deep fryer, oven, cereal maker, candy machine, ice cream machine, and the rest of the customizables too

Incoming5643 commented:
Just gonna say it, no one uses table crafting.

phil235 commented:
>port the deep fryer, oven, cereal maker, candy machine, ice cream machine, and the rest of the customizables too

I considered it, but that's too many snowflake machines for my taste (except the icecream machine).

>no one uses table crafting.
One of this PR's goal is to change that.

Iamgoofball commented:
the machines aren't snowflake anymore they actually were moved to a base cooking machine thing by /vg/station and it looks like you're porting vg's edition of the code

paprka commented:
Those machines are all really bad goofball

They're literally INSERT THING, GET THING NAMED LIKE THING YOU INSERTED

it's really boring and uninspired

ike709 commented:
>no one uses table crafting.
>One of this PR's goal is to change that.

Besides this, :+1:

paprka commented:
table crafting is literally perfectly fine outside of the fact that it uses popups

Iamgoofball commented:
@paprka people enjoyed the shit out of deep frying stuff though

WJohn commented:
Table crafting really ought to be made waaay more usable. It doesn't give you a list of what might potentially be buildable, even if you don't have the materials, and there's no tabs or anything to separate what should be craftable. It's also super inconsistent on what you can and can't use it for. Flamethrowers, assemblies and spears are still done by hand, yet shotguns, IEDs and tech shells are table crafting exclusive. This seriously needs to be addressed soon.

paprka commented:
i think all crafting should be made table crafting, but you should be able to do them on the floor as well by dragging the floor tile onto you, only with twice the build time. Y/n?

WJohn commented:
That's another thing I've been wondering about, how should "dragging sprites onto other sprites" be handled? I mean in all cases. It's used so sparsely and so randomly that it either needs to be applied to a TON more shit or be removed entirely.

paprka commented:
Drag and drop is objectively better than right clicking or verbs, but cases where it's used needs to be more obvious

Jalleo commented:
Tablecrafting isnt used because its a new system technically. (Its not used by coders compared to others)
If somebody made it a interface sorta like alt click show everything on tile it may get more use.

dannno commented:
@iamgoofball
People enjoy shit they can abuse for the short term
In the long term they were bad machines and chef is better off without

WJohn commented:
"New system" that has been in for like 6 months or more.

GunHog commented:
Tablecrafting is not a viable system because only recipes that have all of the requirements are shown on the crafting list. That is user-unfriendly, and is why the primary method of making the same object is preferred to it. Until @Razharas can increase the system's usability, I do not recommend forcing dependency on it.

WJohn commented:
I pointed this out in an earlier comment but yeah, table crafting is incredibly un-user friendly.

phil235 commented:
Tablecrafting could use some improvement indeed, but even in its current form it's already better for cooking recipes than what we have now (magic microwave, load items one by one, one mistake means failure and can break the machine, heavy reliance on wiki guide)

Razharas commented:
Tablecrafting ui upgrading is so easy and so hard at the same time
Easy in part when you just add recipes that arent avalible
Hard in that where to place them, and how to make them show their reqs and shit since its all typebased, to instantiate shit each time shit happens is horrible and autolathe's way is just no

Razharas commented on code/modules/crafting/table.dm:
```diff
@@ -20,18 +20,18 @@
+	for(var/obj/item/weapon/reagent_containers/RC in loc)
+		if((RC.flags & OPENCONTAINER) || (R && (RC.type in R.reqs)))
+			for(var/datum/reagent/A in RC.reagents.reagent_list)
+				table_contents[A.type] += A.volume
```
1 - Why a separate loop, you can handle this all in one pass, instead you decide to loop through all items again just for this case
2 - this is the proc that is supposed to check all the things on the table, and it does it, yet for some reason you are trying to tie in a separate recipe here for seemingly no purpose

phil235 replied:
1 - I'm pretty sure I did this because I thought it was faster to avoid the istype(reagent_containers) check.

2 - the original code is taking the reagents in all reagent_containers but we want to exclude reagent_containers that aren't open containers (you can't just extract a reagent from one snack and use it for a new recipe), except when the snack is already going to be used in our recipe. That's why we need the recipe variable here.

Razharas replied:
All for(each) loops will loop through everything in the location, they will just typecheck to kick out unfitting shit and typecast explicitly the fitting ones, so second loop is pretty much never a good idea

Problem with special cases of containers is that tablecrafting absolutely lacks the idea of discrimination of items based on anything thus trying to make it have it with few lines is just a hack

phil235 committed:
Many fixes. Adding custom pizza slices. Reducing amount of icons used. Adding a changelog.

Razharas commented on code/modules/crafting/table.dm:
```diff
@@ -121,11 +121,13 @@
+				if(istype(B, A))
+					continue deletion_loop
+			Deletion.Remove(B)
+			qdel(B)
```
You made the code do the same thing in 6 lines instead of 4 lines
I dunno if win in less 1 boolean negation per cycle is worth it

phil235 replied:
Actually the initial code cannot handle recipes with more than one element in the parts list. It takes part1 and delete all objects that aren't the same type, then it takes part2 and is fucked because the objects of the same type got deleted and removed from Deletion in the first iteration of the parts loop. Thus the change. (yeah, I know there's currently no recipe using more than one parts, but since parts is a list either we make my change or we make parts a simple var)

phil235 committed:
last fixes I hope:
tiny change to latexballoon (b/c I reorganized is_sharp proc already, so why not)
some fixes suggested by razharas in table.dm
Renamed microwave to microwave oven. Adding efficiency effect (upgrading the machine)
Adding rolling pins to the kitchen vending machine (so the two cooks don't fight.

phil235 committed:
Merge branch 'master' of https://github.com/tgstation/-tg-station into CookingOverhaul

Conflicts:
	code/modules/food&drinks/food/snacks_other.dm
	icons/obj/drinks.dmi

phil235 committed:
Adding a table proc checking whether the table is too crowded to craft things on it.
Two small fixes.

Razharas commented on code/modules/crafting/table.dm:
```diff
@@ -163,4 +174,12 @@
+	var/Item_amount = 0
+	for(var/obj/item/I in loc)
+		Item_amount++
+	if(Item_amount <= 20) //is the table crowded?
```
20 is magic number
Either make it defined somewhere or even better make it per-recipe so some things need more space to make than others

Razharas commented:
Theres also something about some poisonous reagents being impossible to use in this system talk in the feedback thread, what is it?

Tenebrosity commented:
Table crafting would be more intuitive if the table's description told players they could craft on it.

phil235 committed:
Making all meat subtypes of snacks/meat
Adding specific cutlets and raw cutlets for a lot of types of meat.
Enabling reagent transfer from ingredients to resulting dish for foodtablecrafting
Adjusting tablecraft food recipes reagents accordingly.
Adding raw pastry base and pastry base food and including it into certain recipes.

phil235 committed:
Rebalancing vitamin amounts in grown
Fixing some sugar amounts in 2 growns
Fixing error from metastation z2 mapfile.

AlexanderUlanH commented:
If this is going to make it impossible to make burnt messes, the recipe for mystery soup should probably be changed to something else.

phil235 commented:
It's stil possible, you just have to put some food that doesn't have a cooked counterpart in the microwave.

TZK13 commented:
>I considered it, but that's too many snowflake machines for my taste (except the icecream machine).

Would you consider adding the ice cream machine to this PR then?

phil235 commented:
>Would you consider adding the ice cream machine to this PR then?
I think I'll leave it to another PR.

scrdest commented on code/modules/food&drinks/food/snacks_cake.dm:
```diff
@@ -1,139 +1,163 @@
+	slice_path = /obj/item/weapon/reagent_containers/food/snacks/cakeslice/carrot
 	slices_num = 5
-	list_reagents = list("nutriment" = 20, "oculine" = 10, "vitamin" = 5)
+	list_reagents = list("oculine" = 10, "vitamin" = 10)
```
The cake is literally not filling at all? That seems silly. Even if it is vanilla base + carrot add-on, it should at least have 1u of nutriment.

phil235 committed:
Merge branch 'master' of https://github.com/tgstation/-tg-station into CookingOverhaul

Conflicts:
	icons/obj/drinks.dmi

Fixing a couple typos. Making sure all eggs used in tablecraft recipes are boiled

phil235 commented:
Alright this should be ready now. I increased the tablecrafting ingredient limit to 30, fixed some small things, added more details to the changelog. (I looked into adding goofball's icecreamvat but there's a lot of stuff to change so I'm not adding it in this PR)

Cheridan commented on code/modules/food&drinks/recipes/tablecraft/recipes_pastry.dm:
```diff
@@ -0,0 +1,212 @@
+	result = /obj/item/weapon/reagent_containers/food/snacks/muffin/booberry
+
+/datum/table_recipe/chawanmushi
+	name = "Chawanmushi"
```
http://en.wikipedia.org/wiki/Chawanmushi

That's not muffins! :fish_cake:

phil235 replied:
That snack was already listed in the muffin category in snacks_pastry.dm

Cheridan commented on code/modules/food&drinks/recipes/tablecraft/recipes_salad.dm:
```diff
@@ -0,0 +1,44 @@
+	)
+	result = /obj/item/weapon/reagent_containers/food/snacks/salad/validsalad
+
+/datum/table_recipe/monkeysdelight
```
That's not a salad! :leaves: :monkey:

phil235 replied:
It's in a bowl, so it could put in either soup or salad. I know the current sprite looks like a soup but the older sprite didn't look like one so I wasn't sure and I have no clue what real life recipe this is supposed to emulate.

TZK13 replied:
Old sprite return when, in the meantime I'd say it's far more of a soup than it is a salad.

Cheridan commented on code/modules/food&drinks/recipes/tablecraft/recipes_soup.dm:
```diff
@@ -0,0 +1,180 @@
+	)
+	result = /obj/item/weapon/reagent_containers/food/snacks/soup/stew
+
+/datum/table_recipe/spacylibertyduff
```
These bottom two are jellies!

phil235 replied:
It's a bowl with a liquid and ingredients so I put it in that file, I can always make a ////////JELLIES//////// subcatergory.

Cheridan commented on code/modules/hydroponics/grown.dm:
```diff
@@ -338,7 +355,7 @@
 	reagents.add_reagent("morphine", 1 + round(potency / 10, 1))
 	reagents.add_reagent("space_drugs", 1 + round(potency / 10, 1))
-	reagents.add_reagent("vitamin", 1 + round((potency / 10), 1))
+	reagents.add_reagent("vitamin", 1 + round((potency / 25), 1))
```
Make sure this doesn't cause container overflow. At potency 100, the total reagents should add up to no more than the container limit (50 iirc)

Cheridan commented:
Otherwise looks good :+1:

Chocobro commented:
hype
